mwdb/web/src/commons/ui/HexView.js

```diff
@@ -39,46 +39,42 @@ class HexViewNumberRenderer {
     }
 }
 
-export default class HexView extends Component {
-    constructor(props) {
-        super(props);
-        this.numberRenderer = new HexViewNumberRenderer();
-        this.editor = React.createRef();
-        this.originalContent = null;
-        this.hexlifiedContent = null;
-    }
+export default function HexView(props) {
+    const [originalContent, setOriginalContent] = useState(null);
+    const [hexlifiedContent, setHexlifiedContent] = useState(null);
+    let editorRef = React.createRef()
+    let numberRenderer = new HexViewNumberRenderer();
 
-    componentDidUpdate(prevProps) {
-        let editor = this.editor.current.editor;
-        if (this.props.mode !== prevProps.mode) {
-            if (this.props.mode === "raw") this.numberRenderer.detach(editor);
-            else this.numberRenderer.attach(editor);
+    useEffect(() => {
+        if (editorRef.current !== null) {
+            let editor = editorRef.current.editor;
+            editor.gotoLine(1);
+            if (props.mode === "hex") numberRenderer.attach(editor);
         }
-    }
+    }, []);
 
-    componentDidMount() {
-        let editor = this.editor.current.editor;
-        editor.gotoLine(1);
-        if (this.props.mode === "hex") this.numberRenderer.attach(editor);
-    }
+    useEffect(() => {
+        if (editorRef.current !== null) {
+            let editor = editorRef.current.editor;
+            if (props.mode === "hex") numberRenderer.detach(editor);
+            else numberRenderer.attach(editor);
+        }
+    });
 
-    get content() {
+    const content = () => {
```

Comment:
Name should be changed to indicate it is a function
```suggestion
    const getContent = () => {
```